### ISSUE TITLE
more-bundle.t hangs on windows

### ISSUE THREAD

**Issue by chorny** [mojolicious/json-validator#170]
* JSON::Validator version: 3.14
* Perl version: 5.26.0 and others (probably depending on YAML::XS 0.67 availability)
* Operating system: Windows XP

```
Deep recursion on subroutine "JSON::Validator::_resolve" at C:\Strawberry260\cpan\build\JSON-Validator-3.14-0\blib\lib/JSON/Validator.pm line 542.
Deep recursion on subroutine "JSON::Validator::_resolve_ref" at C:\Strawberry260\cpan\build\JSON-Validator-3.14-0\blib\lib/JSON/Validator.pm line 505.
YAML::XS Error: Expected STREAM_START_EVENT; Got: 0 != 1 at C:\Strawberry260\cpan\build\JSON-Validator-3.14-0\blib\lib/JSON/Validator.pm line 358.
(hangs)
```

**Comment by perlpunk** [comment]
@chorny  Just to factor out YAML::XS here - could you possibly test with the newest YAML::XS 0.80?

The error message could point to a memory leak when loading invalid YAML, and I just fixed a leak.

**Comment by chorny** [comment]
Tested with 0.80, same problem

**Comment by jhthorsen** [comment]
Sorry, I don't have windows, so I won't be able to figure out why this doesn't work.

Also seems to only be an issue on certain versions of Windows..? https://app.thorsen.pm/cpantestreport/dist/JSON-Validator-3.14?osname=MSWin32

**Comment by karenetheridge** [comment]
Well that's indicating that a change made in `bundle` (or one of the private methods it uses) didn't work, since the tests in more-bundle.t *used* to cause infinite recursion until the implementation was changed.  Interesting that this only happens on windows! perhaps a difference with how Tie::RefHash is implemented?

**Comment by sudo-batman** [comment]
Just FYI - I also got the similar issue with the new version.

  JSON::Validator version: 3.16
  Perl version: 5.30.0 (Strawberry)
  Operating system: Windows 10

```
t/load-yaml.t .......................... ok
t/more-bundle.t ........................ Deep recursion on subroutine "JSON::Validator::_resolve" at C:\berrybrew\5.30.0_64\cpan\build\JSON-Validator-3.16-2\blib\lib/JSON/Validator.pm line 574.
Deep recursion on subroutine "JSON::Validator::_resolve" at C:\berrybrew\5.30.0_64\cpan\build\JSON-Validator-3.16-2\blib\lib/JSON/Validator.pm line 574.
Deep recursion on subroutine "JSON::Validator::_resolve" at C:\berrybrew\5.30.0_64\cpan\build\JSON-Validator-3.16-2\blib\lib/JSON/Validator.pm line 574.
Deep recursion on subroutine "JSON::Validator::_resolve" at C:\berrybrew\5.30.0_64\cpan\build\JSON-Validator-3.16-2\blib\lib/JSON/Validator.pm line 574.
Deep recursion on subroutine "JSON::Validator::_resolve_ref" at C:\berrybrew\5.30.0_64\cpan\build\JSON-Validator-3.16-2\blib\lib/JSON/Validator.pm line 537.
```

**Comment by jhthorsen** [comment]
Fixed by @charsbar in #180.
